modules/auxiliary/server/capture/ldap.rb

```diff
@@ -0,0 +1,249 @@
+                )
+              end
+              print_good("LDAP Login Attempt => From:#{state[client][:name]} Username:#{state[client][:user]} Password:#{state[client][:pass]}")
+              result_code = Net::LDAP::ResultCodeAuthMethodNotSupported if result_code.nil?
+              service.encode_ldap_response(
+                pdu.message_id,
+                result_code,
+                '',
+                Net::LDAP::ResultStrings[result_code],
+                Net::LDAP::PDU::BindResult
+              )
+            when Net::LDAP::PDU::SearchRequest # search request
+              # Perform query against some loaded LDIF structure
+              filter = Net::LDAP::Filter.parse_ldap_filter(pdu.search_parameters[:filter])
+              attrs = pdu.search_parameters[:attributes].empty? ? :all : pdu.search_parameters[:attributes]
+              res = search_res(filter, pdu.message_id, attrs)
+              if res.nil? || res.empty?
+                result_code = Net::LDAP::ResultCodeNoSuchObject
+              else
+                client.write(res)
+                result_code = Net::LDAP::ResultCodeSuccess
+              end
+              service.encode_ldap_response(
+                pdu.message_id,
+                result_code,
+                '',
+                Net::LDAP::ResultStrings[result_code],
+                Net::LDAP::PDU::SearchResult
+              )
+            when Net::LDAP::PDU::UnbindRequest
+              client.close
+              nil
+            else
+              result_code = Net::LDAP::ResultCodeUnwillingToPerform
+              service.encode_ldap_response(
+                pdu.message_id,
+                result_code,
+                '',
+                Net::LDAP::ResultStrings[result_code],
+                Net::LDAP::PDU::SearchResult
+              )
+            end
+
+      on_send_response(client, res) unless res.nil?
+    rescue StandardError => e
+      on_send_response(client,
+                       service.encode_ldap_response(
+                         1,
+                         Net::LDAP::ResultCodeUnwillingToPerform,
+                         '',
+                         Net::LDAP::ResultStrings[ResultCodeUnwillingToPerform],
+                         Net::LDAP::PDU::BindResult
+                       ))
+      print_error("Failed to handle LDAP request due to #{e}")
+    end
+  end
+
+  def report_cred(opts)
+    service_data = {
+      address: opts[:ip],
+      port: opts[:port],
+      service_name: opts[:service_name],
+      protocol: 'tcp',
+      workspace_id: myworkspace_id
+    }
+
+    credential_data = {
+      origin_type: :service,
+      module_fullname: fullname,
+      username: opts[:user],
+      private_data: opts[:password],
+      private_type: :password,
+      domain: opts[:domain],
+      realm_key: Metasploit::Model::Realm::Key::ACTIVE_DIRECTORY_DOMAIN,
+      realm_value: opts[:domain]
+    }.merge(service_data)
```

Comment:
When no domain is identified this will cause an exception that is swallowed during runtime:
```
[1] pry(#<Msf::Modules::Auxiliary__Server__Capture__Ldap::MetasploitModule>)> e
=> #<ActiveRecord::RecordInvalid: Validation failed: Value can't be blank>
```
Address by suppressing the `realm` keys when not found. Also `domain` is not a processed option for in `create_credential`.
```suggestion
    credential_data = {
      origin_type: :service,
      module_fullname: fullname,
      username: opts[:user],
      private_data: opts[:password],
      private_type: :password,
    }.merge(service_data)
    if opts[:domain]
      credential_data = {
        realm_key: Metasploit::Model::Realm::Key::ACTIVE_DIRECTORY_DOMAIN,
        realm_value: opts[:domain]
      }.merge(credential_data)
    end
```

Reply:
Done